---
title: "Association figures"
author: "Christian Diener"
output:
    html_notebook: default
    html_document:
        toc: true
        toc_float: true
---

# Microbiome phenotype associations

```{r}
library(mbtools)
library(ggplot2)

theme_set(theme_bw())
status_map <- c("healthy", "IFG", "IGT", "IFG+ITG", "T2D", "T2D treated")

ps <- readRDS("../data/taxonomy_clean.rds")
sdata <- sample_data(ps)
sdata$diabetes_status <- factor(status_map[sdata$diabetes_status], levels=status_map)
sample_data(ps) <- sdata

# treatment naive samples
naive <- subset_samples(ps, as.numeric(diabetes_status) < 6 & metformin == 0)
```


## Metformin related changes

Checking for the changes reported earlier.

```{r}
data <- plot_counts(ps, "metformin", "genus", c("Escherichia/Shigella", "Intestinibacter"), 
                    normalized=T, only_data=T)

ggplot(data, aes(x=c("metformin-", "metformin+")[as.numeric(value)], y=reads + 0.5, color=value)) + 
  geom_boxplot() + geom_jitter(width=0.2, alpha=0.5) + guides(color=F) +
  facet_wrap(~ taxa) + scale_y_log10() + labs(x="", y="normalized reads")
ggsave("metformin.svg", width=6, height=4)
```

As you can see only very few samples with metformin treatment. The respective tests.

For Intestinibacter:

```{r}
wilcox.test(reads ~ value, data[taxa == "Intestinibacter"])
```

For E. coli:

```{r}
wilcox.test(reads ~ value, data[taxa == "Escherichia/Shigella"])
```

## Treatment-related changes

```{r, fig.width=3.5, fig.height=5.5}
t2d <- subset_samples(ps, as.numeric(diabetes_status) %in% c(1, 5, 6))

genera <- c("Escherichia/Shigella", "Anaerostipes", "Blautia", "Veillonella", "Romboutsia", "Intestinibacter")
data <- plot_counts(t2d, "diabetes_status", "genus", genera, only_data=T) 
ggplot(data, aes(x=value, y=reads + 0.5, color=value)) + geom_boxplot(outlier.colour=NA) + guides(color=F) +
  geom_jitter(width=0.25, alpha=0.5, size=0.5) + scale_y_log10() + facet_wrap(~ taxa, ncol=2) + 
  labs(x="", y="normalized reads") + theme(axis.text.x = element_text(angle = 30, hjust = 1, vjust=1))
ggsave("treated_untreated.svg", width=3.5, height=5.5)
```

Corresponding tests:

```{r}
tests = data[, .(healthy_vs_T2D = wilcox.test(reads[value == "healthy"],  reads[value == "T2D"])$p.value,
                 healthy_vs_treated = wilcox.test(reads[value == "healthy"],  reads[value == "T2D treated"])$p.value), by=taxa]
tests
```

## Good cop - bad cop

```{r}
tests <- fread("../data/tests_genus.csv")
```

### Diabetes group/status

```{r, fig.width=6, fig.height=6}
genera <- c("Escherichia/Shigella", "Anaerostipes", "Blautia", "Veillonella")
plot_counts(naive, "diabetes_status", "genus", genera, normalized = T) +
  geom_jitter(width=0.2, alpha=0.25, size=1) + geom_smooth(method="glm", aes(group=0, x=as.numeric(value))) +
  labs(x="", y="normalized reads")
ggsave("diabetes_status.svg", width=6, height=6)
```

Corresponding tests:

```{r}
tests[variable == "diabetes_status" & genus %in% genera]
```

### insulin


```{r, fig.width=6, fig.height=4}
sdata <- sample_data(naive)
ggplot(sdata, aes(x=diabetes_status, y=beta_cell_disposition_index, color=diabetes_status)) + 
  geom_hline(yintercept=2, lty="dashed") + geom_boxplot() + geom_jitter(width=0.25, alpha=0.25) +
  labs(x="", y="beta cell disposition")
sdata$beta_cells_affected <- c("normal", "affected")[(sdata$beta_cell_disposition_index < 2) + 1]
ggsave("beta_cell_disposition.svg", width=6, height=6)

sample_data(naive) <- sdata
```

```{r, fig.width=6, fig.height=6}
genera <- c("Escherichia/Shigella", "Anaerostipes", "Blautia", "Veillonella")
data <- plot_counts(naive, "beta_cells_affected", "genus", genera, normalized = T, only_data=T)
ggplot(data[!is.na(value)], aes(x=value, y=reads, color=value)) + geom_boxplot() +
  geom_jitter(width=0.2, alpha=0.25) + facet_wrap(~ taxa) + scale_y_log10() +
  labs(x="", y="normalized reads") + guides(color=F)
ggsave("beta_cell_disposition_index.svg", width=6, height=6)
```

Corresponding tests:

```{r}
tests[variable == "beta_cell_disposition_index" & genus %in% genera]
```

```{r}
data[, .(pval = wilcox.test(reads ~ value)$p.value), by=taxa]
```


### number of risk factors:

```{r, fig.width=6, fig.height=6}
plot_counts(naive, "num_risk_factors", "genus", genera) +
  geom_jitter(width=0.2, alpha=0.25, size=0.5) + geom_smooth(method="glm", aes(group=0, x=as.numeric(value))) +
  labs(x="no. of risk factors", y="normalized reads")
ggsave("num_risk_factors.svg", width=6, height=6)
```


Corresponding tests:

```{r}
tests[variable == "num_risk_factors" & genus %in% genera]
```

### Area under glucose curve

```{r, fig.width=6, fig.height=6}
plot_counts(naive, "auc_glucose", "genus", genera) + geom_smooth(method="glm") +
  labs(x="area under glucose curve", y="normalized reads") + scale_x_log10(breaks=c(2000, 5000, 10000, 20000, 50000))
ggsave("auc_glucose.svg", width=6, height=6)
```

Corresponding tests:

```{r}
tests[variable == "auc_glucose" & genus %in% genera]
```

## General test stats

Significant tests from total tests:

```{r}
paste(nrow(tests[padj < 0.05]), "/", nrow(tests))
```

Genera with most significant tests:

```{r}
by_genus <- tests[, .(nsig=sum(padj < 0.05, na.rm=T)), by="genus"]
by_genus[order(-nsig)][1:10]
```

Significant tests by variable group:

```{r}
annotation <- readxl::read_excel("../data/clinical_new.xlsx",
                                 sheet = "annotations_clean")
annotated_tests <- merge(tests, annotation, by.x="variable", by.y="name")
by_group <- annotated_tests[, .(nsig=sum(padj < 0.05, na.rm=T),
                                ntests=length(padj),
                                nvars=uniqueN(variable)), 
                            by="group"][, "percent" := nsig/ntests]
by_group[order(-nsig)]
```

```{r}
by_genus <- annotated_tests[, .(nsig=sum(padj < 0.05, na.rm=T),
                                ntests=length(padj),
                                nvars=uniqueN(variable)), 
                            by="genus"][, "percent" := nsig/ntests]
by_genus[order(-nsig)]
```

```{r, fig.width=4, fig.height=3}
ggplot(by_group, aes(x=nsig, y=percent, color=group)) + geom_point(size=2) +
  geom_text(aes(label=group), hjust=0, nudge_x=2) + xlim(0, 78) +
  labs(x="significant associations", y="positive test rate") + guides(color=F)
ggsave("sig_plot.svg", width=4, height=3)
```

